### PR TITLE
[12.0][ADD] hr_cae_event_timesheet_sheet

### PR DESCRIPTION
**Pull request by mclaeysb**
Add a module that adds a button on 'done' events which opens a wizard that generates Timesheet sheets for all participants. It suggests timesheet sheet users via the employees, contacts or email addresses in the event's 'attended' registrations, as well as via the event organiser.

Also moved employee_id field on event registration from hr_cae_event_promotion to hr_cae_event.

Attentions points during review: 
- The conditional 'warning' in the wizard
- The addition operator `|=` collecting the users (set union operator) and the three ways possible users are collected.

I also added the OCA repo `timesheet` in the `oca_dependencies.txt`. Remarkably, the CI tests on the other timesheet PR passed even though this repo was not included there yet.

### CODE REVIEW

File: hr_cae_event_timesheet_sheet/models/event.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class Event(models.Model):
+    _inherit = "event.event"
+
+    timesheet_sheet_created = fields.Boolean(
+        string="Timesheet Sheets created before."
```
Comment (robinkeunen) [inline hr_cae_event_timesheet_sheet/models/event.py]: _Timesheet Sheets already before_

File: hr_cae_event_timesheet_sheet/tests/test_event.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Manuel Claeys Bouuaert <manuel@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import tests
+
+
+class TestEvent(tests.common.TransactionCase):
```
Comment (robinkeunen) [inline hr_cae_event_timesheet_sheet/tests/test_event.py:9]: I love reading tests! ;-) 

It's better practice to write more "unitary" tests, that is "one test per scenario". Here for example, it'd be easier to test / fix / read (and better documentation) if you had separate tests for `partner_1`, `partner_3` and `employee` (case in point, I don't understand what  `partner_2` adds to `partner_1`)

```python
def test_create_portal_user_from_registrated_partner(self):
    ...
def test_open_registration_doe_not_log_time_sheet(self):
    ...
def test_create_portal_user_from_registrated_employee(self):
    ...
```

File: hr_cae_event_timesheet_sheet/wizard/event_to_timesheet.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        users |= event.user_id
+        defaults["name"] = event.name
+        defaults["date"] = event.date_begin
+        defaults["duration"] = event.duration
+        defaults["user_ids"] = [(6, 0, users.ids)]
+        defaults["timesheet_sheet_created"] = event.timesheet_sheet_created
+        return defaults
+
+    @api.multi
+    def create_timesheet_sheets(self):
```
Comment (robinkeunen) [inline hr_cae_event_timesheet_sheet/wizard/event_to_timesheet.py]: Shoudn't it be `create_timesheet_line`?  A sheet is the weekly object that collects the lines if I understand that module correctly.

**Review by robinkeunen — CHANGES_REQUESTED** [review]

File: hr_cae_event_timesheet_sheet/wizard/event_to_timesheet.py (hunk 3 above)
Comment (mclaeysb) [reply hr_cae_event_timesheet_sheet/wizard/event_to_timesheet.py]: That would be more correct!

File: hr_cae_event_timesheet_sheet/tests/test_event.py (hunk 2 above)
Comment (mclaeysb) [reply hr_cae_event_timesheet_sheet/tests/test_event.py:9]: I was thinking about this when cycling back home. Will do so! Partner 2 doesn't have the partner_id linked in it's registration.

**Pull request merged**